### DISCUSSION TITLE
New release and an opportunity to contribute

### DISCUSSION THREAD

**Discussion by kelimuttu** [mozilla/multi-account-containers#2672] [Announcements]
Hello everyone!

We are excited to announce that we have just published a new release of the extension. The full release notes can be found [here][rel-notes]. The following is a summary of what to expect:

- [#2663 ](https://github.com/mozilla/multi-account-containers/issues/2263)- This bug fix correctly remembers the choice of the "Always open in container" confirmation page when re-opening a web page that has this setting. Previously, switching to a different container with the "Always open in container" option and remembering your choice would still cause the confirmation page to show up.
- [#303](https://github.com/mozilla/multi-account-containers/issues/303) - With this change, you can clear a container's storage and cookies in its respective panel, and reset a site's cookies through a container's "Manage site list" option by clicking the refresh icon. 
- [#1958](https://github.com/mozilla/multi-account-containers/issues/1958) - The container ordering in the toolbar and the add-on popup now matches.

Have a question for the above? Let us know at the [new Q&A section][discussions] we have opened under discussions. 

I’d like to take this opportunity to invite you all to contribute to this project. There are currently [516 open issues][issues] submitted by community members on Multi-Account Containers (MAC)’s GitHub repository. These issues consist of bugs with the current extension, and feature requests.

A guide to setting up your local repository and starting to contribute can be found [here][contribute].

Tips for contributing:
1. Choose [an issue][issues] that you would like to work on.
2. Fork the repository and follow the instructions for setting it up locally.
3. Run the add-on locally and try reproducing the issue.
4. Debug add-ons by clicking the “Settings” icon in about:addons, and then clicking “Debug Add-ons”
5. Click “Inspect” on the MAC add-on to open developer tools for the popup extension (see [this documentation][extension-doc] for more information)
6. Once you have a fix ready, commit your changes with the following commit message template: “Fix #<insert issue id #>: <short description>”
7. Push your changes and open a pull request for review.

Have any questions? Ask the rest of the community in [our discussions board][discussions]!

[contribute]: https://github.com/mozilla/multi-account-containers/blob/main/CONTRIBUTING.md
[discussions]: https://mzl.la/3SEI8Mq
[extension-doc]: https://extensionworkshop.com/documentation/develop/debugging/
[issues]: https://github.com/mozilla/multi-account-containers/issues
[rel-notes]: https://github.com/mozilla/multi-account-containers/releases/tag/8.2.0

**Comment by dannycolin** [comment]
@achernyakevich-sc Are you still using Multi-Account Containers at your workplace? If yes and there's any interest in helping on the development, feel free to let us know. I'd be also curious to know what features that hasn't been implemented yet would be the most useful for you.

**Comment by Juraj-Masiar** [comment]
If you would like to migrate to TypeScript, I would be happy to help / teach / pair-program and pass all my TS knowledge.
Looking at the codebase, we could convert it completely in a few days. The hardest part would likely be setting up the build system / tests / IDE.

**Reply by kelimuttu** [reply]
Thanks for your interest to help, @Juraj-Masiar I'll pass the idea to our team who should be able to comment on your question with more technical details.

**Reply by groovecoder** [reply]
> If you would like to migrate to TypeScript, I would be happy to help / teach / pair-program and pass all my TS knowledge. Looking at the codebase, we could convert it completely in a few days. The hardest part would likely be setting up the build system / tests / IDE.

Thanks for the offer. At this time, I do not want to migrate the project to TypeScript without a compelling reason. [The top-voted bugs](https://github.com/mozilla/multi-account-containers/issues?q=is%3Aissue+is%3Aopen+label%3Abug+sort%3Areactions-%2B1-desc) are primarily behavior & logic bugs - not the kinds of bugs that TypeScript is likely to fix. And while I personally prefer TypeScript to Javascript, I don't think the effort to set up and maintain the additional build & testing infrastructure is worth it for this project right now.

**Reply by Juraj-Masiar** [reply]
@groovecoder as you wish, but if you change your mind, feel free to contact me in the future :)

Let me just say, I've been programming in JavaScript for 10 years and I'm using TypeScript for about 4 years and I can't imagine working on a project without it (unless it's a single file project).
The benefits in refactoring are immense, you'll catch 90% issues with compiler, not at runtime, so no more `TypeError cannot read undefined`!
And learning curve for existing JavaScript programmers is super fast, since majority of code doesn't change at all (most of the time the types are inferred).
So it's not like it's gonna make development slower :), it's more like having a co-programmer that checks your code and even helps your IDE make refactoring across files!

Anyway, thank you for maintaining this project! It's been a great help.

**Comment by kelimuttu** [comment]
Hi folks, 

Just wanna share here that Andrea Marchesini (Principal Software Engineer) from the Firefox team is going to join us in the community call that we're going to have tomorrow (5 PM UTC) to share more about this contribution opportunity. If you don't have time to join live, you can also submit questions ahead of time by filing out [this form](https://survey.alchemer.com/s3/7947983/5ffe73722df7). Recording will also be available afterward. You can check more info about this call [here](https://support.mozilla.org/forums/contributors/717133#post-88101). 

See you there!

**Comment by jarcoz13** [comment]
hola buenos días te escribo desde MozColombia, desde la comunidad nos gustaria organizar una charla de como contribuir en el proyecto, desde versión nuevos miembros de la comunidad  y nuevos personas que se acerquen al proyecto, de como se puede desarrollar líneas de contacto desde las comunidades de mozilla latam para este mes de octubre
att Alejandro Cortazar 
miembro comunidad MozColombia
mozcolombia@gmail.com
------

Hello, good morning, I am writing to you from MozColombia, from the community, we would like to organize a talk on how to contribute to the project, from new members of the community and new people who approach the project, on how contact lines can be developed from the Mozilla Latin America communities for this month of October.

att Alejandro Cortazar 
member of the MozColombia community

**Reply by kelimuttu** [reply]
Hi Alejandro, we'll keep in touch to discuss the plan with you.
